core/parallel_state_processor.go

```diff
@@ -280,6 +290,14 @@ func (p *ParallelStateProcessor) Process(block *types.Block, statedb *state.Stat
 
 	coinbase, _ := p.bc.Engine().Author(header)
 
+	deps, delayMap := GetDeps(block.Header().TxDependency)
```

Comment:
Can we have some unit tests for block processing with dependencies? The unit tests will allow us to easily test changes in this file without re-syncing a chain.